### PR TITLE
Install Ansible Automation Platform Operator to prod

### LINKED ISSUE #109: Ecological Forecasting: Install Ansible Automation Platform Operator
The [Ecological Forecasting Collaboratory project](https://research.redhat.com/blog/research_project/prototyping-a-distributed-asynchronous-workflow-for-iterative-near-term-ecological-forecasting/) would like to install the `Ansible Automation Platform` Operator to NERC. This will allow us to implement the [Event Driven Automation Portfolio Architecture](https://www.redhat.com/architect/portfolio/detail/27) into the project, which is the main goal of our collaboration. This is also very useful for running Ansible playbooks on virtual machines running on NERC OpenStack. The [Ecological Forecasting project has already started integrating with AAP](https://github.com/computate-org/smartabyar-smartvillage/tree/main/openshift/kustomize/bundles/aap/app) for scheduling ecological forecasting.

### PR DESCRIPTION
**Pull request by computate**
The [Ecological Forecasting Collaboratory project](https://research.redhat.com/blog/research_project/prototyping-a-distributed-asynchronous-workflow-for-iterative-near-term-ecological-forecasting/) would like to install the `Ansible Automation Platform` Operator to NERC. This will allow us to implement the [Event Driven Automation Portfolio Architecture](https://www.redhat.com/architect/portfolio/detail/27) into the project, which is the main goal of our collaboration. This is also very useful for running Ansible playbooks on virtual machines running on NERC OpenStack. The [Ecological Forecasting project has already started integrating with AAP](https://github.com/computate-org/smartabyar-smartvillage/tree/main/openshift/kustomize/bundles/aap/app) for scheduling ecological forecasting.

closes https://github.com/OCP-on-NERC/operations/issues/109

### CODE REVIEW

File: cluster-scope/base/operators.coreos.com/subscriptions/ansible-automation-platform-operator/subscription.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+  name: ansible-automation-platform-operator
+  namespace: aap
```
Comment (dystewart) [inline cluster-scope/base/operators.coreos.com/subscriptions/ansible-automation-platform-operator/subscription.yaml:5]: I think we'll need to add aap namespace and an operatorgroup.

Comment (computate) [reply cluster-scope/base/operators.coreos.com/subscriptions/ansible-automation-platform-operator/subscription.yaml:5]: Thanks for catching that @dystewart , it's fixed in the commit.

**Comment by larsks** [comment]
Minor nitpick: please wrap commit messages at around 72 characters or so. In general, we try to follow [common conventions](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html) for the formatting and content of these messages.

File: cluster-scope/base/core/namespaces/aap/namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  labels:
+    openshift.io/cluster-monitoring: "true"
+  name: aap
```
Comment (larsks) [inline cluster-scope/base/core/namespaces/aap/namespace.yaml:6]: I know that I am going to both misread and mistype this as `app` all the time, but I see that this is the canonical namespace name for this operator so I guess we keep it :smile: .

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml [hunk 3]
```diff
@@ -8,6 +8,9 @@ resources:
 - ../../bundles/acct-mgt
 - ../../bundles/patch-operator
 - ../../bundles/xdmod-reader
+- ../../base/core/namespaces/aap
+- ../../base/operators.coreos.com/operatorgroups/aap
+- ../../base/operators.coreos.com/subscriptions/ansible-automation-platform-operator
```
Comment (larsks) [inline cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: We would typically move this collection of files into a *bundle* in `cluster-scope/bundles` (this allows us to activate/deactivate the operator with a single line in our overlay). See e.g. `cluster-scope/bundles/group-sync-operator`.

**Review by larsks — CHANGES_REQUESTED** [review]

File: cluster-scope/base/core/namespaces/aap/namespace.yaml (hunk 2 above)
Comment (computate) [reply cluster-scope/base/core/namespaces/aap/namespace.yaml:6]: LOL, it does look like a typo :D

File: cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml (hunk 3 above)
Comment (computate) [reply cluster-scope/overlays/nerc-ocp-prod/kustomization.yaml]: Thanks for the feedback @larsks . I learned some new vim tricks and updated my commit. I also made a bundle for ansible-automation-platform-operator.

**Review by jtriley — APPROVED** [review]

**Review by larsks — APPROVED** [review]

**Review by larsks — APPROVED** [review]

**Pull request merged**